src/main/java/org/ohdsi/webapi/shiro/PermissionManager.java

```diff
@@ -96,8 +109,32 @@ public void removeUserFromRole(String roleName, String login, UserOrigin origin)
     UserEntity user = this.getUserByLogin(login);
 
     UserRoleEntity userRole = this.userRoleRepository.findByUserAndRole(user, role);
-    if (userRole != null && (origin == null || origin.equals(userRole.getOrigin())))
+    if (userRole != null && (origin == null || origin.equals(userRole.getOrigin()))) {
+      logger.debug("Removing user from role: {}, {}, {}", user.getLogin(), role.getName(), userRole.getOrigin());
       this.userRoleRepository.delete(userRole);
+    }
+  }
```

Comment:
As we build out the shared artifact owner/reader permissions, we will need to ensure that they are system roles since this and related code removes all non-system roles